pool/chainstate.go

```diff
@@ -190,32 +193,52 @@ func (cs *ChainState) handleChainUpdates(ctx context.Context) {
 			if err != nil {
 				log.Errorf("unable to create header from bytes: %v", err)
 				close(msg.Done)
-				cs.cfg.Cancel()
 				continue
 			}
 
-			// Delete mined work if it is disconnected from the chain. At this
-			// point a mined confirmed block will have its corresponding
-			// accepted block record persisted.
-			id := AcceptedWorkID(header.BlockHash().String(), header.Height)
+			// Check if the disconnected block confirms a mined block, if it
+			// does unconfirm it.
+			id := AcceptedWorkID(header.PrevBlock.String(), header.Height-1)
+			confWork, _ := FetchAcceptedWork(cs.cfg.DB, id)
```

Comment:
I would suggest a comment here explaining why it is acceptable to ignore errors from `FetchAcceptedWork()`